### PR TITLE
create-product-card

### PR DESCRIPTION
**Pull request by Prokopovych16**
- [DEMO LINK](https://Prokopovych16.github.io/layout_product-cards/)
- [TEST REPORT LINK](https://Prokopovych16.github.io/layout_product-cards/report/html_report/)

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -6,8 +6,38 @@
...
+    <link rel="preconnect" href="https://fonts.gstatic.com">
+    <link href="https://fonts.googleapis.com/css2?family=Poppins:wght@400;500;700;800&family=Roboto:wght@400;500;700&display=swap" rel="stylesheet">
   </head>
   <body>
-    <h1>Product cards</h1>
+    <div data-qa="card" class="product-card">
+      <img
+        class="product-card__image"
+        src="images/imac.jpeg"
+        alt="123">
```
Comment (Viktor-Kost) [inline src/index.html:18]: and logo should be wrapped with a link
```suggestion
      <img
        class="product-card__image"
        src="images/imac.jpeg"
        alt="123"
      >
```

File: src/index.html [hunk 2]
```diff
@@ -6,8 +6,38 @@
...
+
+      <h1 class="product-card__h1">APPLE A1419 iMac 27" Retina 5K Monoblock (MNED2UA/A)</h1>
```
Comment (Viktor-Kost) [inline src/index.html:20]: it's better to use `h2` since t's not a page title

File: src/index.html [hunk 3]
```diff
@@ -6,8 +6,38 @@
...
+          <div class="stars__star"></div>
+          <div class="stars__star"></div>
+          <div class="stars__star"></div>
+          <div class="stars__star"></div>
+        </div>
+        <p class="product-card__div-for-stars-p">Reviews: 5</p>
+      </div>
+
+      <div class="product-card__div-price">
+        <h2 class="product-card__div-price-h2">Price:</h2>
```
Comment (Viktor-Kost) [inline src/index.html:36]:
```suggestion
        <p class="product-card__div-price-h2">Price:</p>
```

File: src/styles/index.scss [hunk 4]
```diff
@@ -1,3 +1,126 @@
...
+  padding: 0;
+
+  --color-for-main-text: #060B35;
+  --color-for-secondary-text: #616070;
+  --color-for-button: #00ACDC;
+}
+
+.product-card {
+  max-width: 200px;
+  padding: 33px 19px 16px;
```
Comment (Viktor-Kost) [inline src/styles/index.scss:13]:
```suggestion
  padding: 32px 16px 16px;
```

File: src/styles/index.scss [hunk 5]
```diff
@@ -1,3 +1,126 @@
...
+  box-sizing: border-box;
```
Comment (Viktor-Kost) [inline src/styles/index.scss:14]: move this rule to `*` selector

File: src/index.html (hunk 2 above)
Comment (Viktor-Kost) [reply src/index.html:20]: let's make more meaningful name
```suggestion
      <h1 class="product-card__title">APPLE A1419 iMac 27" Retina 5K Monoblock (MNED2UA/A)</h1>
```

File: src/index.html [hunk 6]
```diff
@@ -6,8 +6,38 @@
...
+      <img
+        class="product-card__image"
+        src="images/imac.jpeg"
+        alt="123">
+
+      <h1 class="product-card__h1">APPLE A1419 iMac 27" Retina 5K Monoblock (MNED2UA/A)</h1>
+
+      <p class="product-card__text-code">Product code: 195434</p>
+
+      <div class="product-card__div-for-stars">
```
Comment (Viktor-Kost) [inline src/index.html:24]:
```suggestion
      <div class="product-card__rating">
```

File: src/index.html [hunk 7]
```diff
@@ -6,8 +6,38 @@
...
+        <div class="stars stars--4">
+          <div class="stars__star"></div>
+          <div class="stars__star"></div>
+          <div class="stars__star"></div>
+          <div class="stars__star"></div>
+          <div class="stars__star"></div>
+        </div>
+        <p class="product-card__div-for-stars-p">Reviews: 5</p>
```
Comment (Viktor-Kost) [inline src/index.html:32]:
```suggestion
        <p class="product-card__reviews">Reviews: 5</p>
```

File: src/index.html [hunk 8]
```diff
@@ -6,8 +6,38 @@
...
+      </div>
+
+      <div class="product-card__div-price">
```
Comment (Viktor-Kost) [inline src/index.html:35]:
```suggestion
      <div class="product-card__price-wrapper">
```

File: src/index.html (hunk 3 above)
Comment (Viktor-Kost) [inline src/index.html:36]:
```suggestion
        <h2 class="product-card__price-text">Price:</h2>
```

File: src/index.html [hunk 9]
```diff
@@ -6,8 +6,38 @@
...
+        <h2 class="product-card__div-price-h2">Price:</h2>
+        <p class="product-card__div-price-p">$2,199</p>
```
Comment (Viktor-Kost) [inline src/index.html:37]:
```suggestion
        <p class="product-card__price-value">$2,199</p>
```

File: src/styles/index.scss [hunk 10]
```diff
@@ -1,3 +1,126 @@
...
+  &__image {
+    width: 160px;
+    height: 134px;
+    margin-bottom: 40px;
+  }
+
+  &__h1 {
+    margin: 0;
+    margin-bottom: 4px;
+    max-width: 166px;
```
Comment (Viktor-Kost) [inline src/styles/index.scss:25]: no needs to hardcode width

File: src/styles/index.scss [hunk 11]
```diff
@@ -1,3 +1,126 @@
...
+    font-size: 12px;
+    font-style: normal;
+    font-weight: 500;
+    line-height: 18px;
+  }
+
+  &__text-code {
+    margin: 0;
+    margin-bottom: 16px;
+    max-width: 98px;
```
Comment (Viktor-Kost) [inline src/styles/index.scss:36]: the same with width
```suggestion
```

File: src/styles/index.scss [hunk 12]
```diff
@@ -1,3 +1,126 @@
...
+    margin-bottom: 40px;
+  }
+
+  &__h1 {
+    margin: 0;
+    margin-bottom: 4px;
+    max-width: 166px;
+    color: var(--color-for-main-text);
+    font-size: 12px;
+    font-style: normal;
```
Comment (Viktor-Kost) [inline src/styles/index.scss:28]: this rule is default, no needs to rereclare. check other cases
```suggestion
```

File: src/styles/index.scss [hunk 13]
```diff
@@ -1,3 +1,126 @@
...
+    font-weight: 700;
+    line-height: 16px; /* 114.286% */
+    text-transform: uppercase;
+  }
+}
+
+.stars {
+  display: flex;
+  height: 16px;
```
Comment (Viktor-Kost) [inline src/styles/index.scss:102]: redundant
```suggestion
```

File: src/styles/index.scss [hunk 14]
```diff
@@ -1,3 +1,126 @@
...
+  box-sizing: border-box;
```
Comment (Viktor-Kost) [inline src/styles/index.scss:103]: this rule we will set at `*`

File: src/styles/index.scss [hunk 15]
```diff
@@ -1,3 +1,126 @@
...
+  background-image: url(../images/star.svg);
+  background-repeat: no-repeat;
+  background-position: center;
+}
+
+.stars__star:not(:last-child) {
+  margin-right: 4px;
+}
+
+.stars.stars--4 > .stars__star:nth-child(-n + 4) {
```
Comment (Viktor-Kost) [inline src/styles/index.scss:118]: reuse styles from `stars` homework. we need to work with all possible modifiers (1-5)

File: src/styles/index.scss [hunk 16]
```diff
@@ -1,3 +1,126 @@
...
+  background-image: url(../images/star-active.svg);
+}
+
+.product-card__button:hover {
```
Comment (Viktor-Kost) [inline src/styles/index.scss:122]: move hover effect as a nesting of `.product-card__button`

File: src/styles/index.scss [hunk 17]
```diff
@@ -1,3 +1,126 @@
...
+      font-size: 16px;
+      font-style: normal;
+      font-weight: 700;
+      line-height: 18px;
+    }
+  }
+
+  &__button {
+    box-sizing: border-box;
+    width: 166px;
```
Comment (Viktor-Kost) [inline src/styles/index.scss:86]: full width
```suggestion
    width: 100%;
```

File: src/styles/index.scss [hunk 18]
```diff
@@ -1,3 +1,126 @@
...
+    height: 40px;
```
Comment (Viktor-Kost) [inline src/styles/index.scss:87]: it's better not hardcode height for elements with content inside. it's better to use padding

File: src/styles/index.scss [hunk 19]
```diff
@@ -1,3 +1,126 @@
...
+  }
+
+  &__text-code {
+    margin: 0;
+    margin-bottom: 16px;
+    max-width: 98px;
+    color: var(--color-for-secondary-text);
+    font-size: 10px;
+    font-style: normal;
+    font-weight: 400;
```
Comment (Viktor-Kost) [inline src/styles/index.scss:40]: this is default value as well

File: src/styles/index.scss [hunk 20]
```diff
@@ -1,3 +1,126 @@
 body {
+  font-family: Roboto, sans-serif;
   margin: 0;
+  padding: 0;
```
Comment (Viktor-Kost) [inline src/styles/index.scss:4]: body doesn't have default padding

File: src/styles/index.scss [hunk 21]
```diff
@@ -1,3 +1,126 @@
...
+  box-sizing: border-box;
+
+  &__image {
+    width: 160px;
+    height: 134px;
+    margin-bottom: 40px;
+  }
+
+  &__h1 {
+    margin: 0;
```
Comment (Viktor-Kost) [inline src/styles/index.scss:23]: you'd better reset margin for `h2` and `p` the same as you did with `body`

File: src/styles/index.scss [hunk 22]
```diff
@@ -1,3 +1,126 @@
...
+      margin: 0;
+      font-size: 16px;
+      font-style: normal;
+      font-weight: 700;
+      line-height: 18px;
+    }
+  }
+
+  &__button {
+    box-sizing: border-box;
```
Comment (Viktor-Kost) [inline src/styles/index.scss:85]:
```suggestion
```

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]

File: src/index.html [hunk 23]
```diff
@@ -6,8 +6,40 @@
...
     <title>Product cards</title>
     <link rel="stylesheet" href="./styles/index.scss">
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com">
+    <link href="https://fonts.googleapis.com/css2?family=Poppins:wght@400;500;700;800&family=Roboto:wght@400;500;700&display=swap" rel="stylesheet">
   </head>
   <body>
-    <h1>Product cards</h1>
+    <div data-qa="card" class="product-card">
+      <a href="123">
```
Comment (volodymyr-soltys97) [inline src/index.html:15]:
```suggestion
      <a href="#">
```

File: src/index.html (hunk 6 above)
Comment (volodymyr-soltys97) [reply src/index.html:26]: Still not fixed

File: src/index.html (hunk 7 above)
Comment (volodymyr-soltys97) [reply src/index.html:34]: The same as above

File: src/styles/index.scss [hunk 24]
```diff
@@ -1,3 +1,124 @@
 body {
+  font-family: Roboto, sans-serif;
   margin: 0;
+
+  --color-for-main-text: #060B35;
+  --color-for-secondary-text: #616070;
+  --color-for-button: #00ACDC;
```
Comment (volodymyr-soltys97) [inline src/styles/index.scss:7]: Create a variable with root selector

File: src/styles/index.scss (hunk 5 above)
Comment (volodymyr-soltys97) [reply src/styles/index.scss:13]: Still not fixed

File: src/styles/index.scss (hunk 10 above)
Comment (volodymyr-soltys97) [reply src/styles/index.scss:24]: The same

File: src/styles/index.scss (hunk 12 above)
Comment (volodymyr-soltys97) [reply src/styles/index.scss:27]: And here

File: src/styles/index.scss [hunk 25]
```diff
@@ -1,3 +1,124 @@
...
+  }
+
+  &__text-code {
+    margin: 0;
+    margin-bottom: 16px;
+    max-width: 98px;
+    color: var(--color-for-secondary-text);
+    font-size: 10px;
+    font-style: normal;
+    font-weight: 400;
```
Comment (volodymyr-soltys97) [inline src/styles/index.scss:39]: Default value, so you can just remove it everywhere
```suggestion
```

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
Demo link doesn't have a styles, need to fix it.
Also need to fix comments from the previous review
image

File: src/styles/index.scss (hunk 5 above)
Comment (Prokopovych16) [reply src/styles/index.scss:21]: I tried to move this code to * selecetor, but all my styles were crashed.

**Review by lerastarynets — CHANGES_REQUESTED** [review]
Prev problems not fixed. Pls ask for help in **fe_chat** if you don't understand something

**Pull request closed without merging**